crates/core/src/host/scheduler.rs

```diff
@@ -258,14 +259,23 @@ impl SchedulerActor {
         }
     }
 
-    async fn handle_queued(&self, id: ScheduledReducerId) {
+    async fn handle_queued(&mut self, id: Expired<ScheduledReducerId>) {
+        let delay = id.deadline().elapsed();
+        let id = id.into_inner();
+        self.key_map.remove(&id);
         let Some(module_host) = self.module_host.upgrade() else {
             return;
         };
         let Some(scheduled) = self.db.get(id.0.to_le_bytes()).unwrap() else {
             return;
         };
         let scheduled: ScheduledReducer = bsatn::from_slice(&scheduled).unwrap();
+        // Note, we are only tracking the time a reducer spends delayed in the queue.
+        // This does not account for any time the executing thread spends blocked by the os.
+        DB_METRICS
+            .scheduled_reducer_delay_ns
+            .with_label_values(&module_host.info().address, &scheduled.reducer)
+            .observe(delay.as_nanos() as f64);
```

Comment:
I know this got talked about, and see that you're using nanos, but I lost the logic for the final conclusion.

It seems to me there are a couple ways to represent this.

One is that you just take the integral number of nanos, stick that in an f64, and move on with your life. So 10 billion nanos would be (10 * 1000 * 1000 * 1000) as f64, and 1 nano would be 1 as f64

The other is that you take treat the integral number of nanos as a fixed point number of seconds, and then convert that to a floating point number of seconds. So 10 billion nanos would be 10 as f64, and 1 nano would be .000000001 as f64.

The former seems much easier (just an `as` away)

Thinking through the fidelity of the f64 number of nanoseconds representation. 2^52 nanos is 52 days. So any single latency we measure is very unlikely to be impacted, but aggregates are (for long enough running modules).

I'm assuming then that this is fine, but wanted to type out the thought.